### DISCUSSION TITLE
Parser gets confused when combining CommandArguments, settings inheritance and branches

### DISCUSSION THREAD

**Discussion by RamonMeffert** [spectreconsole/spectre.console#1309] [General]
**Information**
 - OS: Windows 11
 - Version: 0.47.0
 - Terminal: Powershell 7.3.5 in VS Code (1.80.1) terminal

**Describe the bug**

When inheriting settings, configuring a branch, and using a `CommandArgument` in the base class, the parser gets confused:

<details><summary>Settings and Commands</summary>
<p>

```c#
public class BaseSettings : CommandSettings
{
    [CommandArgument(0, "<ID>")]
    public int Id { get; set; }
}

public class InheritedSettings : BaseSettings
{
    [CommandArgument(0, "<NAME>")]
    public string Name { get; set; }
}

public class BaseCommand : Command<BaseSettings>
{
    public override int Execute([NotNull] CommandContext context, [NotNull] BaseSettings settings)
    {
        Console.WriteLine(settings.Id);
        return 0;
    }
}

public class InheritedCommand : Command<InheritedSettings>
{
    public override int Execute([NotNull] CommandContext context, [NotNull] InheritedSettings settings)
    {
        Console.WriteLine($"{settings.Id}, {settings.Name}");
        return 0;
    }
}
```

</p>
</details> 

```csharp
var app = new CommandApp<BaseCommand>();
		
app.Configure(c => {
    c.AddBranch("name", name => {
        name.AddCommand<InheritedCommand>("set");
    });
});
```

```sh
PS C:\> dotnet run -- 123
123

PS C:\> dotnet run -- 123 name set "Susan"

Error: Unknown command 'name'.

       123 name set "Susan"
           ^^^^ Did you mean 'name'?
```

**To Reproduce**

Minimal example: <https://dotnetfiddle.net/jpVdJ3>

**Expected behavior**

The parser should be able to distinguish branches and commands, I think? I'm not sure what's going wrong.

**Screenshots**

N/A

**Additional context**

The example here is simplified, if you want a more elaborate example (more branches, more complex settings), let me know!

**Comment by nils-a** [comment]
@RamonMeffert The parser is not really confused here, the error-message is... let's say "not really good". 😁 

You created a default command by using `new CommandApp<BaseCommand>()`, so with no other command specified, your app will run the `BaseCommand`. This is why 

```text
PS C:\> dotnet run -- 123
123
```

works the way it does.

Additionally, you created a branch and a command using

```cs
app.Configure(c => {
    c.AddBranch("name", name => {
        name.AddCommand<InheritedCommand>("set");
    });
});
```

So, now, If you run `--help` you'd get:

```text
PS C:\> dotnet run -- name set --help
USAGE:
    test-1.dll name set <ID> <NAME> [OPTIONS]

ARGUMENTS:
    <ID>
    <NAME>

OPTIONS:
    -h, --help    Prints help information
```

which means, that by running 

```ps
PS C:\> dotnet run -- name set 123 "Susan"
```

You'd get the "ouptut" that you wanted. The "mix" of arguments and command-path in your example is currently not supported.
